test/integration/test_integration_rioxarray.py

```diff
@@ -2933,6 +2933,20 @@ def test_rio_write_gcps():
     _check_rio_gcps(darr, gdal_gcps, gcp_crs)
 
 
+def test_rio_write_gcps_no_crs():
+    """
+    Test setting gcps in dataarray, when gcp_crs is not present.
+    """
+    gdal_gcps, _ = _create_gdal_gcps()
+    gcp_crs = None
+
+    darr = xarray.DataArray(1)
+    darr["spatial_ref"] = None
```

Comment:
I don't think this line is needed, correct?
```suggestion
```

Reply:
I had also thought not but it fails when writing the CRS without it. 

```
KeyError: 'spatial_ref'

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)

      darr = xarray.DataArray(1)
      #darr['spatial_ref'] = None
----> darr.rio.write_gcps(gdal_gcps, None)

File ~/modis/rioxarray/rioxarray/rioxarray.py:1238, in XRasterBase.write_gcps(self, gcps, gcp_crs, grid_mapping_name, inplace)
   1234     data_obj = data_obj.rio.write_crs(
   1235         gcp_crs, grid_mapping_name=grid_mapping_name, inplace=inplace
   1236     )
   1237 geojson_gcps = _convert_gcps_to_geojson(gcps)
-> 1238 data_obj.coords[grid_mapping_name].attrs["gcps"] = geojson_gcps
   1239 return data_obj

File /share/apps/anaconda3-2022.05/envs/pangeo/lib/python3.10/site-packages/xarray/core/coordinates.py:386, in DataArrayCoordinates.__getitem__(self, key)
    385 def __getitem__(self, key: Hashable) -> T_DataArray:
--> 386     return self._data._getitem_coord(key)

File /share/apps/anaconda3-2022.05/envs/pangeo/lib/python3.10/site-packages/xarray/core/dataarray.py:819, in DataArray._getitem_coord(self, key)
    817 except KeyError:
    818     dim_sizes = dict(zip(self.dims, self.shape))
--> 819     _, key, var = _get_virtual_variable(self._coords, key, dim_sizes)
    821 return self._replace_maybe_drop_dims(var, name=key)

File /share/apps/anaconda3-2022.05/envs/pangeo/lib/python3.10/site-packages/xarray/core/dataset.py:186, in _get_virtual_variable(variables, key, dim_sizes)
    184 split_key = key.split(".", 1)
    185 if len(split_key) != 2:
--> 186     raise KeyError(key)
    188 ref_name, var_name = split_key
    189 ref_var = variables[ref_name]

KeyError: 'spatial_ref'
```

Reply:
In that case, this won't work how you are expecting. Probably need logic similar to this in `write_gcps`:
https://github.com/corteva/rioxarray/blob/205052675b2e2d070a699aa55359806e894e39ec/rioxarray/rioxarray.py#L588-L634